helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -560,6 +560,29 @@ private void processMaintenanceMode(String clusterName, final boolean enabled,
     }
   }
 
+  private enum ResetPartitionFailureReason {
+    INSTANCE_NOTALIVE("%s is not alive in cluster %s"),
+    INSTANCE_NONEXISTENT("%s does not exist in cluster %s"),
+    RESOURCE_NONEXISTENT("resource %s is not added"),
+    PARTITION_NONEXISTENT("not all %s exist"),
+    PARTITION_NOT_ERROR("%s is NOT found"),
+    STATE_MODEL_NONEXISTENT("%s is NOT found"),
+    PENDING_MSG("a pending message exists: %s");
+
+    private String message;
+
+    ResetPartitionFailureReason(String message) {
+      this.message = message;
+    }
+
+    public String getMessage(String instanceName, String resourceName,
```

Comment:
The method should either return a formated message or a pure format string. The current design is mixing and would be hard to use.